flattentool/cli.py

```diff
@@ -222,19 +230,40 @@ def main():
 
     if args.subparser_name is None:
         parser.print_help()
-    elif args.subparser_name == 'create-template':
-        # Pass the arguments to the create_template function
-        # If the schema file does not exist we catch it in this exception
-        try:
-            # Note: Ensures that empty arguments are not passed to the create_template function
-            create_template(**kwargs_from_parsed_args(args))
-        except (OSError, IOError) as e:
-            print(text_type(e))
-            return
-    elif args.subparser_name == 'flatten':
-        flatten(**kwargs_from_parsed_args(args))
-    elif args.subparser_name == 'unflatten':
-        unflatten(**kwargs_from_parsed_args(args))
+    else:
+        def handler(type, value, traceback):
```

Comment:
Would prefer this to be a top level function as well, possibly a new module for warnings customisations.

Reply:
The closure is needed here (the handler refers to `args`). If moved, we'd need a method that returns a method, which seems to increase complexity without clear benefit.